packages/block-editor/src/components/link-control/index.js

```diff
@@ -219,20 +255,52 @@ function LinkControl( {
 			const isInternal = startsWith( val, '#' );
 			const isTel = protocol.includes( 'tel' );
 
-			const handleManualEntry =
+			const maybeURL =
 				isInternal ||
 				isMailto ||
 				isTel ||
 				isURL( val ) ||
 				( val && val.includes( 'www.' ) );
 
-			return handleManualEntry
+			return maybeURL
 				? handleDirectEntry( val, args )
 				: handleEntitySearch( val, args );
 		},
 		[ handleDirectEntry, fetchSearchSuggestions ]
 	);
 
+	const handleOnCreate = async ( entityTitle ) => {
+		let newEntity;
+
+		setIsResolvingLink( true );
+		setErrorMsg( null );
+		try {
+			newEntity = await createEntity( 'page', entityTitle );
```

Comment:
Why do we hard-code `'page'` here? Couldn't we just pass through the regular `suggestion` object? If it needs to be a page for the Navigation Block, handle that in the navigation block.

Reply:
> Why do we hard-code 'page' here?

I'm glad you raised this. I did highlight this as one of my concerns in my code review above. Maybe you missed it - GH is weird sometimes!

I agree entirely with what you say. I'm going to amend this.